### PR TITLE
When user logsout there is an error

### PR DESCRIPTION
**Pull request by Elkrival**
Refactored logout and signout functionality to use a callback function as the error states.

### CODE REVIEW

File: server/utils/passport/ensure-authenticated.js [hunk 1]
```diff
@@ -13,12 +13,18 @@ export default async function ensureAuthenticated(req, res, next) {
...
-
         res.clearCookie('connect.sid')
 
+        req.logout(function (err) {
+          if (err) return next(err)
+
+          req.session.destroy(function (sessionError) {
+            if (sessionError) return next(sessionError)
+          })
+        })
```
Comment (mikestone14) [inline server/utils/passport/ensure-authenticated.js]: we run this code block in 3 different files, which makes me think we should extract it and DRY it up.

File: views/forms/UserProfileForm/UserProfileFormFields.jsx [hunk 2]
```diff
@@ -36,6 +36,7 @@ const UserProfileFormFields = ({ control }) => {
         name="mail"
         size="small"
         fullWidth
+        disabled
```
Comment (mikestone14) [inline views/forms/UserProfileForm/UserProfileFormFields.jsx:39]: we just removed this in the last PR, are we sure we want to add it back?

Comment (JackVCurtis) [reply views/forms/UserProfileForm/UserProfileFormFields.jsx:39]: Yeah we had to un-disable it to fix a data issue in staging.

File: server/utils/passport/ensure-authenticated.js (hunk 1 above)
Comment (Elkrival) [reply server/utils/passport/ensure-authenticated.js]: [133cd90](https://github.com/AMP-SCZ/dpdash/pull/579/commits/133cd90acec446de23055fd47c4cbb3f790c5897)

File: server/strategies/logout.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export const logout = (req, res, next) => {
```
Comment (JackVCurtis) [inline server/strategies/logout.js]: This isn't exactly a strategy in the passport.js sense - could we move this function to the `utils` directory?

Comment (Elkrival) [reply server/strategies/logout.js]: Good catch
[89f8b42](https://github.com/AMP-SCZ/dpdash/pull/579/commits/89f8b42c8be681b87e2497243b9c27e31680d918)

**Review by JackVCurtis — APPROVED** [review]
👍

**Pull request merged**